tests/testthat/test-EventStudy.R

```diff
@@ -1,4 +1,25 @@
 
+# Housekeeping ------------------------------------------------------------
+
+test_that("does not modify input data (even if input is data.table)", {
```

Comment:
This is the reason we are copying the data, correct? Interesting solution.

Is a `data.table` passed to a function always modified in the function? Are we sure this is a concern? It'd be preferable to avoid the copying if possible, just for efficiency.

Reply:
Thanks @zhizhongpu! Great idea to add the test.